modules/core/src/smithy4s/ByteArray.scala

```diff
@@ -18,6 +18,8 @@ package smithy4s
 
 import java.util.Base64
 
+@deprecated("Use smithy4s.Blob instead", since = "0.18.0")
```

Comment:
Thought: if it's deprecated and we're replacing all its usages, chances are people will get breakage because of all the type mismatches. At this point I'd say remove the type altogether

Reply:
agreed